### PR TITLE
continuous-delivery: Added tls support, improved health checking and support for solanoci

### PR DESCRIPTION
**Pull request by emaildanwilson**
@brendandburns you mentioned I should ping you on updates I make to this.

### CODE REVIEW

**Comment by bprashanth** [comment]
@kubernetes/sig-testing

**Comment by emaildanwilson** [comment]
@bprashanth this isn't related to the test infrastructure for Kubernetes and is instead for the use by people that intend to deploy their own apps on their own kubernetes clusters. Please see this previous pull request for more details. https://github.com/kubernetes/contrib/pull/215#issuecomment-156839985

**Comment by bprashanth** [comment]
I'm guessing sig-testing would be interested in anything Ci/Cd

**Comment by bgrant0607** [comment]
cc @kubernetes/examples

File: continuousdelivery/deploy/ensure-kubectl.sh [hunk 1]
```diff
@@ -49,11 +49,11 @@ if [ ! -e ~/.kube ]; then
 fi
 
 if [ ! -e ~/.kube/kubectl ]; then
-    wget https://storage.googleapis.com/kubernetes-release/release/v1.0.6/bin/linux/amd64/kubectl -O ~/.kube/kubectl
+    wget https://storage.googleapis.com/kubernetes-release/release/v1.1.4/bin/linux/amd64/kubectl -O ~/.kube/kubectl
```
Comment (pwittrock) [inline continuousdelivery/deploy/ensure-kubectl.sh]: Would you update this to 1.2.1 now that it is out?

**Comment by pwittrock** [comment]
@emaildanwilson How were these tested?  Is there anyway for me to test short of running them on solano, circleci, jenkins?

File: continuousdelivery/deploy/ensure-kubectl.sh (hunk 1 above)
Comment (bgrant0607) [inline continuousdelivery/deploy/ensure-kubectl.sh]: 1.2.2 :-)

**Comment by emaildanwilson** [comment]
These are only testable on solano, circleci and jenkins currently. I'd be happy to work with someone to add in any other CI and/or more tests. If there is a way to set this up to deploy this service to a real kubernetes cluster during the standard "Travis CI" that is used let me know. I tested these on my own setup of each CI against multiple k8s clusters and have several different R&D teams using them for their deployments which results in most of these updates. The whole point of these scripts is so people using kubernetes can orchestrate deployments to multiple clusters using their own CI system. At least until we have ubernetes :).

File: continuousdelivery/deploy/deploy-service.sh [hunk 2]
```diff
@@ -29,50 +29,80 @@ $DIR/ensure-kubectl.sh
 
 #set config context
 ~/.kube/kubectl config use-context ${CONTEXT}
+~/.kube/kubectl version
 
-#get user password and api ip from config data
-export kubepass=`(~/.kube/kubectl config view -o json | jq ' { mycontext: .["current-context"], contexts: .contexts[], users: .users[], clusters: .clusters[]}' | jq 'select(.mycontext == .contexts.name) | select(.contexts.context.user == .users.name) | select(.contexts.context.cluster == .clusters.name)' | jq .users.user.password | tr -d '\"')`
+#get user, password, certs, namespace and api ip from config data
+export kubepass=`(~/.kube/kubectl config view -o json --raw --minify  | jq .users[0].user.password | tr -d '\"')`
```
Comment (pwittrock) [inline continuousdelivery/deploy/deploy-service.sh]: nice

**Pull request merged**